youcan-shop/youcan-ui
# [chore]: Remove `histoire` in favour of Storybook

## Reason

After careful evaluation of `histoire` as our solution of choice for handling component stories, it was very clear that the solution lacked maturity and isn't production ready in cases where we'll want to open up the stories to our internal team.

This lack can be measured with the following:

- A layout that is hard to navigate and use
- Many bugs on startup
- Infant community
- Not an ideal DX compared to competitors

After some thinking we went for [Storybook](https://storybook.js.org/)

## QA Steps

The existing stories were very much lacking, with that in mind + the upcoming YouCan Design system in the horizon, only a subset of those stories were put it place as a reference when the refactor is in place.

- [ ] Node version for this project was updated to `v19.2.0`, use `nvm` to side load different node version in the same machine and use the one mentioned above.
- [ ] Run `pnpm i` (note: it might take some time).
- [ ] Next try and run `pnpm story` to visualise the stories we have available.

Next to the deprecation of `histoire`, we updated the version of several packages that we have in place. To ensure everything is working as expected please follow:
- [ ] Run `pnpm dev` to see a page with all the components we have, interact with them all and see if they are working as epected
- [ ] Run `pnpm build` to check if the build phase of the project is also working, we should only export 3 files: `index.css, index.umd.js & index.es.js`
